test/e2e/pod.go

```diff
@@ -94,34 +84,32 @@ var _ = ginkgo.Describe("Pod to external server PMTUD", func() {
 			}
 			e2epod.NewPodClient(f).CreateSync(context.TODO(), clientPod)
 
-			ginkgo.By("Creating the server pod with hostNetwork:true")
-			// Create the server pod.
-			// Wait for 1 minute and if the pod does not come up, select a different port and try again.
-			// Wait for a max of 5 minutes.
-			gomega.Eventually(func() error {
-				serverPodPort = rand.Intn(echoServerPodPortMax-echoServerPodPortMin) + echoServerPodPortMin
-				serverPodName = fmt.Sprintf(echoServerPodNameTemplate, serverPodPort)
-				framework.Logf("Creating server pod listening on TCP and UDP port %d", serverPodPort)
-				serverPod = e2epod.NewAgnhostPod(f.Namespace.Name, serverPodName, nil, nil, nil, "netexec",
-					"--http-port",
-					fmt.Sprintf("%d", serverPodPort),
-					"--udp-port",
-					fmt.Sprintf("%d", serverPodPort))
-				serverPod.ObjectMeta.Labels = map[string]string{
-					"app": serverPodName,
-				}
-				serverPod.Spec.HostNetwork = true
-				serverPod.Spec.NodeName = serverPodNodeName
-				e2epod.NewPodClient(f).Create(context.TODO(), serverPod)
-
-				err := e2epod.WaitTimeoutForPodReadyInNamespace(context.TODO(), f.ClientSet, serverPod.Name, f.Namespace.Name, 1*time.Minute)
-				if err != nil {
-					e2epod.NewPodClient(f).Delete(context.TODO(), serverPod.Name, metav1.DeleteOptions{})
-					return err
-				}
-				serverPod, err = e2epod.NewPodClient(f).Get(context.TODO(), serverPod.Name, metav1.GetOptions{})
-				return err
-			}, 5*time.Minute, 1*time.Second).Should(gomega.Succeed())
+			ginkgo.By("Creating the external server")
+			serverPodPort = rand.Intn(echoServerPodPortMax-echoServerPodPortMin) + echoServerPodPortMin
+			serverPodName = fmt.Sprintf(echoServerPodNameTemplate, serverPodPort)
+			framework.Logf("Creating server pod listening on TCP and UDP port %d", serverPodPort)
+			agntHostCmds := []string{"netexec", "--http-port", fmt.Sprintf("%d", serverPodPort), "--udp-port", fmt.Sprintf("%d", serverPodPort)}
+			externalIpv4, externalIpv6 := createClusterExternalContainer(serverPodName, agnhostImage,
+				[]string{"--network", "kind", "-P", "--cap-add", "NET_ADMIN"},
+				agntHostCmds,
+			)
+
+			if isIPv4Supported() {
+				serverNodeInternalIPs = append(serverNodeInternalIPs, externalIpv4)
+			}
+
+			if isIPv6Supported() {
+				serverNodeInternalIPs = append(serverNodeInternalIPs, externalIpv6)
+			}
+
+			gomega.Expect(len(serverNodeInternalIPs)).To(gomega.BeNumerically(">", 0))
```

Comment:
ah nice `gomega.BeNumerically` ideally we don't do dualstack lane for CP today so its gotta be 1 always but this is better for the future.